### ISSUE TITLE
Backend / Generate JSON not working

### ISSUE THREAD

**Issue by florianmulatz** [JonasJoKuJonas/homeassistant-WebUntis#154]
### The problem

Hi!

As soon as I try to enable "Generate JSON" in Backend Settings it says 

![image](https://github.com/user-attachments/assets/04e0528a-3ede-43d5-ba6c-1f7e9d03bcf3)

in the frontend.

Thx for any help

### Version of Home Assistant?

Core 2024.10.1 Supervisor 2024.10.0 Operating System 13.1 Frontend 20241002.2

### What type of installation are you running?

Home Assistant OS

### Version of WebUntis

v1.2.5

### Traceback/Error logs

```2024-10-10 06:15:14.352 ERROR (SyncWorker_50) [webuntis] {'jsonrpc': '2.0', 'id': '2024-10-10 06:15:14.332332', 'error': {'message': 'no right for getTeachers()', 'code': -8509}}
2024-10-10 06:15:14.357 WARNING (SyncWorker_50) [homeassistant.helpers.frame] Detected that custom integration 'webuntis' calls hass.config_entries.async_update_entry from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassconfig_entriesasync_update_entry at custom_components/webuntis/__init__.py, line 932: self._hass.config_entries.async_update_entry(self._config, options=new_options), please report it to the author of the 'webuntis' custom integration
2024-10-10 06:15:14.358 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry MaxMUSTERMANN for webuntis
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/remote.py", line 123, in _parse_result
    return result_body[u'result']
           ~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'result'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/webuntis/__init__.py", line 854, in get_lesson_json
    for teacher in lesson.teachers
                   ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/misc.py", line 30, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
                                           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/objects.py", line 344, in teachers
    return self._session.teachers(from_cache=True).filter(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/misc.py", line 156, in inner
    data = self._request(jsonrpc_method, jsonrpc_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/session.py", line 124, in _request
    data = rpc_request(self.config, method, params or {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/remote.py", line 91, in rpc_request
    return _parse_result(request_body, result_body)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/remote.py", line 125, in _parse_result
    _parse_error_code(request_body, result_body)
  File "/usr/local/lib/python3.12/site-packages/webuntis/utils/remote.py", line 143, in _parse_error_code
    raise exc
webuntis.errors.RemoteError: no right for getTeachers()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/webuntis/__init__.py", line 61, in async_setup_entry
    await server.async_update()
  File "/config/custom_components/webuntis/__init__.py", line 242, in async_update
    await self._async_status_request()
  File "/config/custom_components/webuntis/__init__.py", line 340, in _async_status_request
    self.next_class = await self._hass.async_add_executor_job(self._next_class)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/webuntis/__init__.py", line 547, in _next_class
    self.next_class_json = self.get_lesson_json(lesson)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/webuntis/__init__.py", line 858, in get_lesson_json
    self.exclude_data_("teachers")
  File "/config/custom_components/webuntis/__init__.py", line 932, in exclude_data_
    self._hass.config_entries.async_update_entry(self._config, options=new_options)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2074, in async_update_entry
    self.hass.verify_event_loop_thread("hass.config_entries.async_update_entry")
  File "/usr/src/homeassistant/homeassistant/core.py", line 462, in verify_event_loop_thread
    frame.report_non_thread_safe_operation(what)
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 229, in report_non_thread_safe_operation
    report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 163, in report
    _report_integration(what, integration_frame, level, error_if_integration)
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 201, in _report_integration
    raise RuntimeError(
RuntimeError: Detected that custom integration 'webuntis' calls hass.config_entries.async_update_entry from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassconfig_entriesasync_update_entry at custom_components/webuntis/__init__.py, line 932: self._hass.config_entries.async_update_entry(self._config, options=new_options). Please report it to the author of the 'webuntis' custom integration.```

### Additional information

_No response_

**Comment by florianmulatz** [comment]
Did not change with current version v1.3.0

**Comment by user45876** [comment]
Same here. This is the error:

`RuntimeError: Detected that custom integration 'webuntis' calls hass.config_entries.async_update_entry from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassconfig_entriesasync_update_entry at custom_components/webuntis/__init__.py, line 1019: self._hass.config_entries.async_update_entry(self._config, options=new_options). Please report it to the author of the 'webuntis' custom integration.`

**Comment by queeek** [comment]
are you using HACS 2 or 1? In my setup with HACS 1, Webuntis 1.2.5 HA Core 2024.9.1 its working fine.

**Comment by florianmulatz** [comment]
I'm using HACS 2 but I'm not sure why this is relevant for the backend feature of WebUntis? 
cheers mate

**Comment by user45876** [comment]
I think it might be related to Core 2024.10 rather - if this is the case, more people will experience the issue.
@florianmulatz When you first hit the issue, did you already have Core 2024.10? I am a first time user and installed 1.3.0 on top of 2024.10.1.

The error is relatively clear, it means that the webuntis custom integration is trying to update from a thread that's not the main event loop thread. Unfortunately I do not have the skills to fix it :-/

`[...]'webuntis' calls hass.config_entries.async_update_entry from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt.[...]`

and 

`[...] For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassconfig_entriesasync_update_entry at custom_components/webuntis/__init__.py, line 1019: self._hass.config_entries.async_update_entry(self._config, options=new_options). Please report it to the author of the 'webuntis' custom integration. [...]`

**Comment by JonasJoKuJonas** [comment]
Quick Fix: add teachers to exclude data

This should actually happen automatically if you have no rights to query the teachers name. But in the new Core 2024.10 this function can no longer be called from a thread and ends in an Error. (This feature should prevent the no rights error from spamming the logs)

**Comment by user45876** [comment]
Thanks for the quick workaround, Jonas, can confirm that this helps. Just want to say that the error does not only spam logs but make the integration unable to run ;)

**Comment by JonasJoKuJonas** [comment]
> Just want to say that the error does not only spam logs but make the integration unable to run ;)

I know. I was talking about the no rights error. The integration is trying to save that it don't has access to the teachers name, but in the newer version you can't save the data at this particular point at the code. So the integration just stops working

**Comment by florianmulatz** [comment]
> Quick Fix: add teachers to exclude data
> 
> This should actually happen automatically if you have no rights to query the teachers name. But in the new Core 2024.10 this function can no longer be called from a thread and ends in an Error. (This feature should prevent the no rights error from spamming the logs)

this worked for me as a quickfix as well! thanks for that indeed!

**Comment by JonasJoKuJonas** [comment]
fixed in v1.3.2
